web/src/store.js

```diff
@@ -25,6 +29,61 @@ export default new Vuex.Store({
     },
   },
   actions: {
+    async reloadApiKey({ state, commit, getters }) {
+      const { user } = getters;
+      const { status, data } = await state.girderRest.get(
+        'api_key', {
+          params: {
+            userId: user._id,
+            limit: 50,
+            sort: 'name',
+            sortdir: 1,
+          },
+        },
+      );
+
+      const [dandiKey] = data.filter(key => key.name === 'dandicli');
+      if (status === 200 && dandiKey) {
+        // send the key id to "PUT" endpoint for updating
+        const { data: { key } } = await state.girderRest.put(`api_key/${dandiKey._id}`);
+        commit('setApiKey', key);
+      }
+    },
+    async fetchApiKey({ state, commit, getters }) {
+      const { user } = getters;
+      const { status, data } = await state.girderRest.get(
+        'api_key', {
+          params: {
+            userId: user._id,
+            limit: 50,
+            sort: 'name',
+            sortdir: 1,
+          },
+        },
+      );
+
+      const [dandiKey] = data.filter(key => key.name === 'dandicli');
+      if (status === 200 && dandiKey) {
+        // if there is an existing api key
+
+        // set the user key
+        commit('setApiKey', dandiKey.key);
+      } else {
+        // create a key using "POST" endpoint
+        const { status: createStatus, data: { key } } = await state.girderRest.post('api_key', null, {
+          params: {
+            name: 'dandicli',
+            scope: JSON.stringify(['core.data.read', 'core.data.write']),
+            tokenDuration: 30,
+            active: true,
+          },
+        });
+
+        if (createStatus === 200) {
+          commit('setApiKey', key);
+        }
+      }
+    },
```

Comment:
It seems to me like these 2 actions overlap a lot could be combined into a  single "fetchApiKey" action that either gets or creates.

Maybe something to look at some other time.